### ISSUE TITLE
no_std support

### ISSUE THREAD

**Issue by cole-miller** [cole-miller/rust-hashsplit#2]
This is kind of a silly thing to worry about right now, but the library probably isn't going to use anything that isn't in `core` and it would be nice to make that independence concrete.

**Comment by cole-miller** [comment]
- [x] Switch to `core::prelude`
- [x] Handle usage of `Vec`
- [x] Support substituting `alloc`

**Comment by cole-miller** [comment]
The `Rolling` type's use of a ring buffer pretty much requires dynamic allocation (well, that or integer generics, which are not available yet — though maybe something could be hacked up with a macro). So I think we'll have a feature that permits compiling against `core` only, giving up `Rolling` but keeping the rest of the crate, including the `Hasher` trait.

**Comment by zenhack** [comment]
Fwiw, now that we've fixed the size of the window as 64, there's no need to have the size configurable in this library either (this was part of the motiviation) -- so presumably you could dodge this by just defining a constant `WINDOW_SIZE = 64` and using that instead of a dynamically allocated vector.

**Comment by cole-miller** [comment]
I did define such a constant (most recent commit on the develop branch), but hadn't made the connection with this issue -- thanks! A number of other things will still require `alloc` but moving `Rolling` out of that category is a big improvement.

**Comment by cole-miller** [comment]
No-alloc `Rolling` is implemented in 98774e0. I'll leave the issue open in case there are more opportunities for this kind of thing.

**Comment by cole-miller** [comment]
`Delimited` and the wrapper `WithRolling` are no-alloc now. `Splits` plainly must allocate. `Spans` doesn't need to allocate itself but extracting it from `ResumableChunk` (which uses `Cow`) would be gross and it's hard to see how it would be *useful* without allocation, so not touching it for now.
